data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java

```diff
@@ -257,9 +266,61 @@ private void onSubscriberFailure(final Throwable failure,
     incrementEventCount(response, recordContext);
     recordDispatchLatency(response, recordContext);
 
-    dlsSender.apply(recordContext.getRecord())
-      .onSuccess(v -> onDeadLetterSinkSuccess(recordContext, finalProm))
-      .onFailure(ex -> onDeadLetterSinkFailure(recordContext, ex, finalProm));
+    // enhance event with extension attributes prior to forwarding to the dead letter sink
+    ConsumerRecordContext transformedRecordContext = errorTransform(recordContext, response);
```

Comment:
I'd prefer using `final` as much as possible, to express objects are not mutable.

See other classes for examples. That would be really my only "nit" comment on the code :smile:

Reply:
ok, no problem. i mainly use golang and kotlin nowadays, almost forgot how to talk in java :) kotlin has nice val/var for this purpose